modules/flowable-spring-boot/flowable-spring-boot-starters/flowable-spring-boot-autoconfigure/src/main/java/org/flowable/spring/aot/FlowableBeanFactoryInitializationAotProcessor.java

```diff
@@ -0,0 +1,304 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flowable.spring.aot;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.ibatis.javassist.util.proxy.ProxyFactory;
+import org.apache.ibatis.scripting.defaults.RawLanguageDriver;
+import org.apache.ibatis.scripting.xmltags.XMLLanguageDriver;
+import org.apache.ibatis.type.TypeHandler;
+import org.flowable.bpmn.converter.BpmnXMLConverter;
+import org.flowable.bpmn.model.ImplementationType;
+import org.flowable.bpmn.model.ServiceTask;
+import org.flowable.cmmn.converter.CmmnXmlConverter;
+import org.flowable.common.engine.api.query.Query;
+import org.flowable.common.engine.impl.db.ListQueryParameterObject;
+import org.flowable.common.engine.impl.de.odysseus.el.ExpressionFactoryImpl;
+import org.flowable.common.engine.impl.persistence.cache.EntityCacheImpl;
+import org.flowable.common.engine.impl.persistence.entity.ByteArrayRef;
+import org.flowable.common.engine.impl.persistence.entity.Entity;
+import org.flowable.common.engine.impl.persistence.entity.EntityManager;
+import org.flowable.common.engine.impl.persistence.entity.TablePageQueryImpl;
+import org.flowable.eventregistry.impl.db.SetChannelDefinitionTypeAndImplementationCustomChange;
+import org.flowable.variable.api.types.VariableType;
+import org.flowable.variable.service.impl.InternalVariableInstanceQueryImpl;
+import org.flowable.variable.service.impl.QueryVariableValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.beans.factory.aot.BeanFactoryInitializationAotContribution;
+import org.springframework.beans.factory.aot.BeanFactoryInitializationAotProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackages;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Josh Long
+ * @author Joram Barrez
+ */
+class FlowableBeanFactoryInitializationAotProcessor implements BeanFactoryInitializationAotProcessor {
+
+    private final PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
```

Comment:
Nit pick comment; ignoring it will not hurt my feelings.

There are 31 occurrences of loggers that use `getClass()`  and are not `static`.

There are 415 that use the form: `public static final Logger LOGGER = LoggerFactory.getLogger(xxxx.class);`, where the variable is uppercase and `static` and explicitly mentions the class.   I would assume consistency would mean that new code should use this later format.

Reply:
I actually prefer the use of a final field for this. It allows to inherit the class and when logging to see the actual instance of the class that is doing the logging. I actually think that we should update our `public static final Logger` to `protected final Logger`